gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -461,15 +472,35 @@ private void buildKafkaMetricReporter(Properties properties) {
       return;
     }
 
-    KafkaReporter.Builder<?> builder = formatEnum.reporterBuilder(this.metricContext);
+    if(topicMetrics.or(topicDefault).isPresent()) {
```

Comment:
Consider have a boolean shouldBuild() method  inside MetricReporterBuilder and EventsReproterBuilder instead of conditional logic in this class.